packages/udt/src/udt/index.ts

```diff
@@ -1,306 +1,394 @@
+    lock: ccc.Script,
+    amount: ccc.FixedPoint,
+    extraData: ccc.BytesLike = [],
+  ): Promise<number> {
+    const outputData = ccc.hexFrom(
+      ccc.bytesConcat(mol.Uint128.encode(amount), extraData),
+    );
+    const cellOutput = ccc.CellOutput.from(
+      {
+        lock,
+        type: this.script,
+      },
+      outputData,
+    );
 
-    return ssri.ExecutorResponse.new(undefined);
+    const cell = { cellOutput, outputData };
+    if ((await this.infoFrom(client, undefined, cell)).balance !== amount) {
+      throw new Error("Internal Error: amount mismatch");
+    }
+
+    return tx.addOutput(cellOutput, outputData);
   }
 
-  /**
-   * Transfers UDT to specified addresses.
-   * @param tx - Transfer on the basis of an existing transaction to achieve combined actions. If not provided, a new transaction will be created.
-   * @param transfers - The array of transfers.
-   * @param transfers.to - The receiver of token.
-   * @param transfers.amount - The amount of token to the receiver.
-   * @returns The transaction result.
-   * @tag Mutation - This method represents a mutation of the onchain state and will return a transaction object.
-   * @example
-   * ```typescript
-   * const { script: change } = await signer.getRecommendedAddressObj();
-   * const { script: to } = await ccc.Address.fromString(receiver, signer.client);
-   *
-   * const udt = new Udt(
-   *   {
-   *     txHash: "0x4e2e832e0b1e7b5994681b621b00c1e65f577ee4b440ef95fa07db9bb3d50269",
-   *     index: 0,
-   *   },
-   *   {
-   *     codeHash: "0xcc9dc33ef234e14bc788c43a4848556a5fb16401a04662fc55db9bb201987037",
-   *     hashType: "type",
-   *     args: "0x71fd1985b2971a9903e4d8ed0d59e6710166985217ca0681437883837b86162f"
-   *   },
-   * );
-   *
-   * const { res: tx } = await udtTrait.transfer(
-   *   signer,
-   *   [{ to, amount: 100 }],
-   * );
-   *
-   * const completedTx = udt.completeUdtBy(tx, signer);
-   * await completedTx.completeInputsByCapacity(signer);
-   * await completedTx.completeFeeBy(signer);
-   * const transferTxHash = await signer.sendTransaction(completedTx);
-   * ```
-   */
-  async transfer(
-    signer: ccc.Signer,
-    transfers: {
-      to: ccc.ScriptLike;
-      amount: ccc.NumLike;
-    }[],
-    tx?: ccc.TransactionLike | null,
-  ): Promise<ssri.ExecutorResponse<ccc.Transaction>> {
-    let resTx;
-    if (this.executor) {
-      const txReq = ccc.Transaction.from(tx ?? {});
-      await txReq.completeInputsAtLeastOne(signer);
-
-      const res = await this.executor.runScriptTry(
-        this.code,
-        "UDT.transfer",
-        [
-          txReq.toBytes(),
-          ccc.ScriptVec.encode(transfers.map(({ to }) => to)),
-          ccc.mol.Uint128Vec.encode(transfers.map(({ amount }) => amount)),
-        ],
-        {
-          script: this.script,
-        },
-      );
-      if (res) {
-        resTx = res.map((res) => ccc.Transaction.fromBytes(res));
-      }
+  async updateOutput(
```

Comment:
Definitely practical method for updating an output's balance!

Reply:
Do you have any particular issue in mind?

Reply:
No. I don't think this part has any issues. Just want you to know that I think it's a good idea to have this.